### PR TITLE
feat: Create last-updated-partial field for research-output

### PR DESCRIPTION
**Pull request by peterstarling**
see: https://trello.com/c/QNJQsmTQ/1169-return-research-outputs-last-updated-timestamp-to-fe

### CODE REVIEW

File: packages/squidex/schema/schemas/research-outputs.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
       "label": "Research Outputs",
       "validateOnPublish": false
     },
-    "scripts": {},
+    "scripts": {
+      "update": "const modified = Object.keys(ctx.data)\n        .filter(key => JSON.stringify(ctx.data[key]) !== JSON.stringify(ctx.oldData[key]))\n        .filter(key => key !== 'publishDate' && key !== 'adminNotes');\n        \nif (modified.length > 0) {\n    ctx.data.test2.iv = JSON.stringify(modified);\n    ctx.data.lastUpdatedPartial.iv = new Date().toISOString().split('.')[0]+\"Z\";\n    replace();\n}\n\n"
```
Comment (peterstarling) [inline packages/squidex/schema/schemas/research-outputs.json:14]:
```typescript
const modified = Object.keys(ctx.data)
  .filter(
    (key) => JSON.stringify(ctx.data[key]) !== JSON.stringify(ctx.oldData[key])
  )
  .filter((key) => key !== "publishDate" && key !== "adminNotes");

if (modified.length > 0) {
  ctx.data.test2.iv = JSON.stringify(modified);
  ctx.data.lastUpdatedPartial.iv = new Date().toISOString().split(".")[0] + "Z";
  replace();
}
```

Comment (johnytiago) [reply packages/squidex/schema/schemas/research-outputs.json:14]: Can we remove the test bit?
Also, I don't think you need to remove the miliseconds from the date. https://support.squidex.io/t/implemented-support-iso-8601-missing-miliseconds/2191

File: apps/asap-server/src/entities/research-output.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export const parseResearchOutput = (
   publishDate: output.data.publishDate?.iv,
   addedDate: output.data.addedDate?.iv,
   tags: output.data.tags?.iv || [],
+  lastUpdatedPartial:
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/asap-server/src/entities/research-output.ts:18]: I'm not sure actually, these are two different fields, what if at some point in our admin panel we want to show when the resource was actually modified - including changes to `adminNotes` and `publishDate`? I know right now we don't have an admin panel (well its squidex) but I would prefer to plan ahead for things like this.

To me this is a whole new field. If for now we want to change it in the UI - then let's make that replacement in the UI but BE should imho show these two separately.

File: packages/squidex/schema/schemas/research-outputs.json [hunk 3]
```diff
@@ -213,6 +215,23 @@
           "isRequiredOnPublish": false,
           "isHalfWidth": false
         }
+      },
+      {
+        "name": "lastUpdatedPartial",
+        "isHidden": false,
+        "isLocked": false,
+        "isDisabled": false,
```
Comment (johnytiago) [inline packages/squidex/schema/schemas/research-outputs.json:223]:
```suggestion
        "isDisabled": true,
```

Just missed this bit. The field should not be editable, right?

**Pull request merged**